beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -631,3 +638,116 @@ func (bs *Server) validateEquivocation(blk interfaces.ReadOnlyBeaconBlock) error
 	}
 	return nil
 }
+
+// GetBlockRoot retrieves the root of a block.
+func (bs *Server) GetBlockRoot(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetBlockRoot")
+	defer span.End()
+
+	var err error
+	var root []byte
+	blockID := mux.Vars(r)["block_id"]
+	if blockID == "" {
+		http2.HandleError(w, "block_id is required in URL params", http.StatusBadRequest)
+		return
+	}
+	switch blockID {
+	case "head":
+		root, err = bs.ChainInfoFetcher.HeadRoot(ctx)
+		if err != nil {
+			http2.HandleError(w, "Could not retrieve head root: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if root == nil {
+			http2.HandleError(w, "No head root was found", http.StatusNotFound)
+			return
+		}
+	case "finalized":
+		finalized := bs.ChainInfoFetcher.FinalizedCheckpt()
+		root = finalized.Root
+	case "genesis":
+		blk, err := bs.BeaconDB.GenesisBlock(ctx)
+		if err != nil {
+			http2.HandleError(w, "Could not retrieve genesis block: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if err := blocks.BeaconBlockIsNil(blk); err != nil {
+			http2.HandleError(w, "Could not find genesis block: "+err.Error(), http.StatusNotFound)
+			return
+		}
+		blkRoot, err := blk.Block().HashTreeRoot()
+		if err != nil {
+			http2.HandleError(w, "Could not hash genesis block: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		root = blkRoot[:]
+	default:
+		isHex := strings.HasPrefix(blockID, "0x")
+		if isHex {
+			blockIDBytes, err := hexutil.Decode(blockID)
+			if err != nil {
+				http2.HandleError(w, "Could not decode block ID into bytes: "+err.Error(), http.StatusBadRequest)
+				return
+			}
+			blockID32 := bytesutil.ToBytes32(blockIDBytes)
```

Comment:
I think the safest option is to do a length check here before calling the bytes32 and throw an error if it doesn't otherwise it's going to pad it.